java/src/jmri/jmrix/ncemonitor/NcePacketMonitorPanel.java

```diff
@@ -480,11 +479,31 @@ void enableDisableWhenOpen(boolean isOpen) {
     void openPortButtonActionPerformed(java.awt.event.ActionEvent e) {
         //log.info("Open button pushed");
         // can't change this anymore
-        String openStatus = openPort((String) portBox.getSelectedItem(), validModelValues[modelBox.getSelectedIndex()], "JMRI");
-        if (openStatus != null) {
-            log.debug("Open Returned: {}", openStatus);
-            return;
+        String portName = (String) portBox.getSelectedItem();
+        int modelValue = validModelValues[modelBox.getSelectedIndex()];
+        int numDataBits = modelBitValues[modelValue];
+        int numStopBits = modelStopValues[modelValue];
+        int parity = modelParityValues[modelValue];
+        int baudrate = modelBaudRates[modelValue];
+        activeSerialPort = AbstractSerialPortController.activatePort(
+                null, portName, log, numStopBits, AbstractSerialPortController.Parity.getParity(parity));
+
+        activeSerialPort.setNumDataBits(numDataBits);
+        activeSerialPort.setBaudRate(baudrate);
+
+        // set RTS high, DTR high
+        activeSerialPort.setRTS(); // not connected in some serial ports and adapters
+        activeSerialPort.setDTR(); // pin 1 in DIN8; on main connector, this is DTR
+
```

Comment:
Perhaps I've missed it, but this new code doesn't seem to be purging the input and setting the `serialStream` and `ostream` variables.

Reply:
Thanks! I had missed setting the streams. That's fixed now.

`AbstractSerialPortController.activatePort()` purges the input stream so I thought that it wasn't necessary to do it in NcePacketMonitorPanel as well. At line 164, `activatePort()` calls `purgeStream(serialPort.getInputStream());`